README.md

```diff
@@ -1,42 +1,106 @@
+
+**META-INF/services/com.exactpro.sf.externalapi.codec.IExternalCodecFactory**
+
+with the content equals to the fully-qualified class name of your factory implementation.
+
+_If you have several implementations of that interface their fully-qualified names should be written in that file each one on the new line._
+
 
 # Configuration
 
-Codec has got four types of connection: stream and general for encode and decode functions.
+Codec has four types of connection: stream and general for encode and decode functions.
 
-* stream encode / decode connections are used for work 24 / 7
-* general encode / decode connections are used for requests on demand
+* stream encode / decode connections works 24 / 7
+* general encode / decode connections works on demand
 
-Messages in stream and general connections are never mixed. 
+Codec never mixes messages from the _stream_ and the _general_ connections. 
 
 Decoding can work in two different modes:
-+ CUMULATIVE (default) - all raw messages in batch will be joined together and decoded. After decoding, content and count of the decoded messages will be compared with the original messages in the batch.
-+ SEQUENTIAL - each message in the batch will be decoded as separate message.
++ **CUMULATIVE** (default) - all raw messages in batch will be joined together and decoded. After decoding, content and count of the decoded messages will be compared with the original messages in the batch.
++ **SEQUENTIAL** - each message in the batch will be decoded as separate message.
 
 This setting can be overridden in a custom config for the application using `decodeProcessorType` parameter.
 
-## Requried pins
+## Bootstrap parameters
 
-Every type of connections is presented two subscribe and publish pins. 
-Configuration should include at least one at a time pins for every types (minimal number of them is 8)
+These parameters specify the codec to be used for the messages decoding/encoding and the mode which should be used.
+Their should be defined in the `custom-config` section of the component configuration.
 
+```yaml
+codecClassName: fully.qualified.class.name.for.Factory
+decodeProcessorType: CUMULATIVE
+```
+
+## Codec implementation parameters
+
+These parameters will be passed to the actual codec implementation to configure its behavior.
+A codec might not have any parameters to configure. In that case, you can omit adding those parameters.
+
+The codec implementation parameters should be located in the container's `/home` directory and stored in the file named `codec_config.yml`.
+It has simple key-value format in YAML.
+```yaml
+---
+param1: value1
+param2: value2
+```
+The set of parameters depends on the codec implementation is used.
+
+## Required pins
+
+Every type of connection has two `subscribe` and `publish` pins.
+The first one is used to receive messages to decode/encode. The second one - to send decoded/encoded messages further.
```

Comment:
The first one is used to receive messages to decode/encode **while the** second one **is used** to send decoded/encoded messages further.